### DISCUSSION TITLE
RX_FSK.ino -  Compiles and uploads fine but blank screen

### DISCUSSION THREAD

**Discussion by wrevenaugh** [dl9rdz/rdz_ttgo_sonde#406] [TTGO firmware]
Using Arduino IDE to upload code.  Having no problems compiling and uploading, and then using the ESP32 Sketch Data Upload tool - everything goes well.  However, once completed, display is blank, green LED flashes on and off, and the RDZSonde Access Point doesn't start/appear in wifi networks.  What am I doing wrong ?   I am assuming RX_FSK is all I need, correct?
I know the board and display work fine (brand new) and everything works fine when I upload a .bin with ESP32 Flash Tool.

The reason I want to use the IDE is I want to modify code to output Imperial Units.  If you know of an Imperial version that exists, please let me know where I might look.

Thank you again for all your hard work!  Can't tell you how many days/hours of enjoyment your awesome software has provided. I have 3 Sondes now, 2-RS41's and 1 DFM17.    
Thanks Again!  Bill

**Comment by dl9rdz** [comment]
From your description, it seems that everything should be right. 

You could look at the serial console in Arduino IDE to possibly get a better understanding what is going wrong.

If you upload the console output here, note that it contains the WiFi passwords that you have configured (in case you changed the default configuration). You can also send it to me to have a look by email (dl9rdz@darc.de)

**Reply by wrevenaugh** [reply]
Thank you so much for your reply.  I've attached my serial monitor output here for you.  I really appreciate it.  Again, I can't begin to tell you how much fun I've had with this thing and how much I appreciate you sharing this wonderful tool with us.  It's just amazing.  Last week I built a 433mhz "Moxon" antenna for it and picked up signals from a Sonde 280 miles (450Km) away.  It was only 2.5 degrees up from the horizon!  What a blast! Thanks again, (is it Hans?) 
[SerialMonitor.txt](https://github.com/dl9rdz/rdz_ttgo_sonde/files/13537040/SerialMonitor.txt)
Bill.

**Reply by dl9rdz** [reply]
Alright, so something is causing the watchdog timer to reset the board. Should not happen.

Looks like a standard TTGO V2.1_1.6 board with an LCD display, right?

What versions of libraries are you using? I have not updated the build process of the binary images, they are using rather old versions: GFX Library for Arduino: v1.1.5, and U8g2 v2.29.11. You could test if downgrading the library to that version in ArduinoIDE fixes the problem (in that case, please let me know the version that you have been using)

**Reply by wrevenaugh** [reply]
Correct, standard TTGO V2.1_1.6 with OLED (that's what you meant right?, not LCD)   https://www.lilygo.cc/products/lora3  433mhz.  I went ahead and tried downgrading U8g2 and GFX Lib to your versions and I still get the same flashing LED and blank screen.  Also, I've attached a new Serial Output as the one I attached previously was from a computer on the new 2.2.1 Arduino IDE (which I realize won't work with your software) so I was actually doing all the work on another computer with the old 1.8.19 IDE.  Could that be the problem? I forget which IDE version I need to use.  
[NewSerial.txt](https://github.com/dl9rdz/rdz_ttgo_sonde/files/13537221/NewSerial.txt)

**Reply by wrevenaugh** [reply]
SOLVED!  **IT WAS THE CORD!** LOL  I changed to a different USB cord and it worked.  I then went back and updated the U8g2 lib to current (2.34.22) and it worked as well, and then went back and updated the GFX for Arduino Lib to current and that DID NOT work (just FYI) so went back to GFX 1.1.5 and all was good.  Thank you again SO much for your responses.  Now all I need to do is figure out where in your code to look for the VS HS and Altitude.  Thank you again!  Bill

**Comment by dl9rdz** [comment]
Also, note that there is still an open issue (https://github.com/dl9rdz/rdz_ttgo_sonde/issues/355). I could add imperial units, but I am still waiting for someone to answer the questions I raised in that issue (in particular, what would be a "good" behavior of the distance display on the TTGO screen, e.g. always miles? switch to feet for shorter distances? I never use imperial units and have no idea what would be a "reasonable" thing to do)

**Reply by wrevenaugh** [reply]
I will address this immediately!  You will save me A LOT of time and headaches if you have a version partially completed with Imperial Units.....  I am going through your code and it is absolutely WAY over my head - you are amazing, man!  :)  Bill

**Reply by wrevenaugh** [reply]
P.S. I would be HAPPY to try and make the conversions myself, but I just can't figure out where best (in your code) to start.  I can't locate the point in the code where you push the HS, VS and Alt to the screen - it is there that I would just do the conversions, but as I said, I can't figure out where you're doing that, lol.
